torch/quantization/fx/_equalize.py

```diff
@@ -489,21 +506,13 @@ def convert_eq_obs(
             inp_quant_obs_node = node.args[0]
             prev_node = inp_quant_obs_node.args[0]
 
-            # TODO: Possible special handling for connected linear layers
-            # We might need to replace the InputEqualizationObserver with another mul operator...
-            # if prev_node.op == 'call_module' and modules[prev_node.target].dtype != torch.float32:
-            # If this is a connecting linear layer, we want to remove the InputEqualizationObserver
-            #     # Replace the current node with the previous node
-            #     orig_users = list(node.users.keys())
-            #     for user_node in orig_users:
-            #         user_node.replace_input_with(node, prev_node)
-            #     # Erase the node
-            #     model.graph.erase_node(node)
-            #     continue
-
             # Update the following input quantization observer's min/max values
             scale_input_observer(node, modules)
 
+            if node_supports_equalization(prev_node, modules):
```

Comment:
can we add some comments here about why we are removing nodes here?

Reply:
Yes, I will add a comment about this. Essentially, if the previous node is a layer that needs to be equalized, then we will remove the current node (input equalization observer) because we do not need to add any equalization nodes between two layers that need to be equalized.

Before: `linear1 (prev_node) -> output_quant_obs1 (inp_quant_obs_node) -> input_eq_obs2 (node) -> linear2`
After: `linear1 (prev_node) -> output_quant_obs1 (inp_quant_obs_node) -> linear2`